README.md

```diff
@@ -1,86 +1,125 @@
+installing binaries directly or building from source code.
+
+In the following, instruction of how to build and install MIGraphX are described with Ubuntu as the OS
+(Instructions of installations on other Linux OSes will come later). Note that all the following instructions are based 
+on that ROCm has been installed successfully. ROCm installation instructions are explained in the [ROCm installation
+guide](https://rocmdocs.amd.com/en/latest/Installation_Guide/Installation-Guide.html).
+
+## Installing from binaries
+With ROCm installed correctly, MIGraphX binaries can be installed on Ubuntu with the following command:
+```
+sudo apt update && sudo apt install -y migraphx
+```
+then the header files and libs are installed under `/opt/rocm-<version>`, where `<version>` is the rocm versoin.
+
+## Building from source
+
+There are two ways to build the MIGraphX sources. One is using the ROCm build tool 
+[rbuild](https://github.com/RadeonOpenCompute/rbuild) to install the prerequisites and
+build the libs with just one command. The other is installing the prerequisites, then using 
+cmake to build the source. In the following, we will first list the prerequisites required to 
+build MIGraphX source code, then describe each of the two approaches.
+
+### List of prerequisites
+The following is a list of prerequisites required to build MIGraphX source. 
 
-## Prerequisites
 * [ROCm cmake modules](https://github.com/RadeonOpenCompute/rocm-cmake) **required**
 * [MIOpen](https://github.com/ROCmSoftwarePlatform/MIOpen) for running on the GPU
+* [rocBLAS](https://github.com/ROCmSoftwarePlatform/rocBLAS) for running on the GPU
 * [HIP](https://github.com/ROCm-Developer-Tools/HIP) for running on the GPU
 * [Protobuf](https://github.com/google/protobuf) for reading [onnx](https://github.com/onnx/onnx) files
 * [Half](http://half.sourceforge.net/) - IEEE 754-based half-precision floating point library
 * [pybind11](https://pybind11.readthedocs.io/en/stable/) - for python bindings
+* [JSON](https://github.com/nlohmann/json) - for model serialization to json string format
+* [MessagePack](https://msgpack.org/index.html) - for model serialization to binary format
 
-## Installing the dependencies
+#### Use the ROCm build tool [rbuild](https://github.com/RadeonOpenCompute/rbuild).
 
-Dependencies can be installed using the ROCm build tool [rbuild](https://github.com/RadeonOpenCompute/rbuild).
-
-To install rbuild:
+In this approach, we need to install the [rbuild](https://github.com/RadeonOpenCompute/rbuild) first, then use it to
+build MIGraphX. rbuild can be installed as (sudo may be needed):
 ```
-pip install https://github.com/RadeonOpenCompute/rbuild/archive/master.tar.gz
+pip3 install https://github.com/RadeonOpenCompute/rbuild/archive/master.tar.gz
 ```
+and pip3 can be installed as `sudo apt update && sudo apt install -y python3-pip`.
+We also need to install [rocm-cmake](https://github.com/RadeonOpenCompute/rocm-cmake), rocblas, and miopen-hip as 
```

Comment:
I think this should be mentioned before installing rbuild. Something:

> To install the rocm dependencies on ubuntu from repo.radeon.com, you can run this `apt` command:
> 
>     sudo apt install rocm-cmake rocblas miopen-hip python3-pip

Then next mention how to install rbuild.